README.rst

```diff
@@ -35,6 +35,6 @@ cd ukrdict
 ```
 3. Install ukrdict package
 ```bash
-setup.py install
+python setup.py install
```

Comment:
Never ever use this. Always use pip.

Reply:
Need use 
```
-e git+https://github.com/serhii73/ukrdict.git#egg=ukrdict
```
Right?

Reply:
> Never ever use this. Always use pip.

Hi, @webknjaz ! Initially this was done by me. As this was the first thing I have packaged, looks like I did it in a wrong way. But could you please explain why `python setup.py install` should never be used? Thanks in advance!

Reply:
@serhii73 yes, that is right or alternatively, use `pip install -e .` if it's already checked out on your computer.

Now, since this is in the installation section and is not targeting devs/contributors it should be even just `ukrdict`. `pip install -e ...` should only be used for development as it just creates a symlink in the file system w/o actually copying the files (except for the metadata only).

Reply:
@undef1nd historically, `setup.py install` was used a lot ~10 yrs ago. Back then, people tended to use more of `distutils`, not setuptools. `distutils` doesn't put the list of files into the metadata so if you install a dist like this, `pip` won't be able to find out which files to delete when you do `pip uninstall` — old versions report "successful" deletion, but it only removed the metadata dir, not the files because it didn't know how to find them. Newer pip versions now show a warning that not all of the things are removed.

Pip itself handles tracking files when you use it, so you're protected from such situations. Also, Pip is a front-end tool for managing dependencies, there's absolutely no reason to use a different, low-level thing for one case and another tool for other cases.

But yeah, the official PyPA recommendation is to avoid invoking `setup.py install`. It's also a leaky abstraction because users don't really need to know how to build the dist, they should only know how to install it with a generic well-known tool.

Nowadays, there's also PEP517+PEP518 which are loosely implemented in the ecosystem which also allows you to abstract away the build system implementation details from the package manager.
Here's how you can make use of it: https://github.com/sanitizers/octomachinery/blob/5e1da30/pyproject.toml#L1-L8
 — this example still uses Pip + setuptools, but they are on the separate sides of the toolchain. This allows you to substitute the build system with anything else (setuptools, flit, poetry or even your own build scripts) while using Pip for invoking the whole thing in a unified manner. You can also replace the front-end part (Pip) with other tools which know how to read the build-system section from `pyproject.toml`.

Reply:
@webknjaz, thanks you so much for the explanation! Your advice on using pip makes way more sense to me now. I definitely need to read more on packaging/installation and the tooling.

Reply:
https://twitter.com/pganssle/status/1152694289920679936?s=19